packages/server/src/user/__tests__/user.service.spec.ts

```diff
@@ -0,0 +1,311 @@
+import { Project, User } from '@prisma/client';
+import { PrismaService } from '../../prisma/prisma.service';
+import { UserService } from '../user.service';
+import { UserTestUtil } from './utils/user.test.util';
+import { addMinutes, isAfter, isBefore, isEqual, isPast, subDays, subHours, subMinutes } from 'date-fns';
+import * as bcrypt from 'bcrypt';
+import * as randomstring from 'randomstring';
+import { NotFoundError } from '@prisma/client/runtime';
+
+describe('UserModule Integration Test (service)', () => {
+  let userTestUtil: UserTestUtil;
+
+  let prisma: PrismaService;
+  let userService: UserService;
+
+  let dummyProjects: Project[];
+  let dummyAdmins: User[];
+  let dummyUsers: User[];
+
+  let randomProject: Project;
+  let randomUser: User;
+
+  const HASH_ROUNDS = 10;
+
+  beforeAll(async () => {
+    userTestUtil = new UserTestUtil();
+
+    const moduleRef = await userTestUtil.setup();
+
+    prisma = moduleRef.get(PrismaService);
+
+    dummyProjects = await userTestUtil.createDummyProjects();
+    dummyAdmins = await userTestUtil.createDummyAdmins();
+    dummyUsers = await userTestUtil.createDummyUsers();
+
+    // create controller and service objects
+    userService = new UserService(prisma);
+  });
+
+  afterAll(async () => {
+    await userTestUtil.tearDown();
+  });
+
+  beforeEach(async () => {
+    randomProject = dummyProjects[Math.floor(Math.random() * dummyProjects.length)];
+    randomUser = dummyUsers.concat(dummyAdmins)[Math.floor(Math.random() * (dummyAdmins.length + dummyUsers.length))];
+  });
+
+  describe('Test cases for createUser()', () => {
+    let createDate: Date;
+    let tempUserPasseword: string;
+    let tempUserInput: any;
+
+    let userCreated: User;
+
+    let createTempUser: boolean;
+
+    beforeEach(async () => {
+      createDate = new Date();
+      tempUserPasseword = randomstring.generate(Math.floor(Math.random() * (64 - 16) + 16));
+      tempUserInput = { projectId: randomProject.id, username: 'temp', email: 'temp@mail.com', password: tempUserPasseword };
+
+      createTempUser = true;
+    });
+
+    afterEach(async () => {
+      if (!createTempUser) return;
+
+      expect(userCreated.projectId).toEqual(randomProject.id);
```

Comment:
These are something handled by the DB, and we didn't write this logic, so this isn't something we need to test.